### PR TITLE
chore: add flakybot

### PR DESCRIPTION
**Pull request by quartzmo**
Adapted from [google-cloud-go/kokoro/continuous.sh](https://github.com/googleapis/google-cloud-go/blob/main/internal/kokoro/continuous.sh#L168-L173).

closes: #1321

### CODE REVIEW

File: internal/kokoro/test.sh [hunk 1]
```diff
@@ -56,3 +56,10 @@ if [[ $KOKORO_JOB_NAME == *"continuous"* ]]; then
 else
     go test -race -v -short ./... 2>&1 | tee $KOKORO_ARTIFACTS_DIR/$KOKORO_GERRIT_CHANGE_NUMBER.txt
 fi
+
+if [[ $KOKORO_BUILD_ARTIFACTS_SUBDIR = *"continuous"* ]]; then
```
Comment (quartzmo) [inline internal/kokoro/test.sh:60]: Do we only want to run flakybot on `continuous`, or should we always run it in this file, meaning for `go test -race -v -short ...` as well?

File: internal/kokoro/test.sh [hunk 2]
```diff
@@ -56,3 +56,10 @@ if [[ $KOKORO_JOB_NAME == *"continuous"* ]]; then
...
+  chmod +x $KOKORO_GFILE_DIR/linux_amd64/flakybot
+  $KOKORO_GFILE_DIR/linux_amd64/flakybot -logs_dir=$GOCLOUD_HOME \
+    -repo=googleapis/google-api-go-client \
+    -commit_hash=$KOKORO_GITHUB_COMMIT_URL_google_api_go_client
```
Comment (quartzmo) [inline internal/kokoro/test.sh:64]: Is `KOKORO_GITHUB_COMMIT_URL_google_api_go_client` valid in this context? I adapted this from [google-cloud-go](https://github.com/googleapis/google-cloud-go/blob/main/internal/kokoro/continuous.sh#L168-L173).

File: internal/kokoro/test.sh (hunk 1 above)
Comment (codyoss) [reply internal/kokoro/test.sh:60]: just continuous.

File: internal/kokoro/test.sh (hunk 2 above)
Comment (codyoss) [reply internal/kokoro/test.sh:64]: I believe it should be!

**Review by codyoss — APPROVED** [review]
Thanks!

**Pull request merged**